### PR TITLE
lockdown - breakout making globalThis properties non-writable

### PR DESCRIPTION
**Pull request by kumavis**
this is in preperation for using LavaMoat which does `lockdown-run.js` but not `lockdown-more.js`

### CODE REVIEW

File: app/scripts/lockdown-more.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+     * @param {Object} descriptor - The propertyName descriptor to check.
+     * @returns {boolean} Whether the propertyName descriptor has any accessors.
+     */
+    function hasAccessor(descriptor) {
+      return 'set' in descriptor || 'get' in descriptor;
+    }
+  })();
+} catch (error) {
+  console.error('Protecting intrinsics failed:', error);
+  if (globalThis.sentry && globalThis.sentry.captureException) {
```
Comment (darkwing) [inline app/scripts/lockdown-more.js:86]: Can we do `if (globalThis.sentry?.captureException) {`?

**Comment by darkwing** [comment]
Naming is difficult but could we make `-more` more specific?  I guess I don't have a better word off the top of my head :/

File: app/scripts/lockdown-more.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+// Make all "object" and "function" own properties of globalThis
+// non-configurable and non-writable, when possible.
+// We call the a property that is non-configurable and non-writable,
```
Comment (rekmarks) [inline app/scripts/lockdown-more.js:3]:
```suggestion
// We call a property that is non-configurable and non-writable,
```

File: app/scripts/lockdown-more.js (hunk 1 above)
Comment (rekmarks) [reply app/scripts/lockdown-more.js:86]:
```suggestion
  if (globalThis?.sentry.captureException) {
```

**Review by rekmarks — CHANGES_REQUESTED** [review]

**Comment by rekmarks** [comment]
@darkwing @kumavis how about `lockdown-intrinsics`?

**Comment by kumavis** [comment]
i definitely struggled to find an appropriate name

**Comment by kumavis** [comment]
we also considered expanding this beyond intrinsics

**Review by rekmarks — APPROVED** [review]
LGTM!

**Review by ryanml — APPROVED** [review]

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**